datamol/mol.py

```diff
@@ -86,28 +87,54 @@ def to_mol(
     return _mol
 
 
-def same_mol(mol1: Optional[Chem.rdchem.Mol], mol2: Optional[Chem.rdchem.Mol]):
+def same_mol(
+    mol1: Optional[dm.Mol],
+    mol2: Optional[dm.Mol],
+    use_non_standard_inchikey: bool = False,
+) -> bool:
     """Check two molecules are the same by comparing their InChiKey.
 
     Invalid molecules (None) are always considered as not the same.
 
     Args:
         mol1: A molecule.
         mol2: A molecule.
+        use_non_standard_inchikey: Whether to use the standard or non-standard InChiKey.
     """
 
     if mol1 is None or mol2 is None:
         return False
 
-    return dm.to_inchikey(mol1) == dm.to_inchikey(mol2)
+    if use_non_standard_inchikey:
+        return dm.to_inchikey_non_standard(mol1) == dm.to_inchikey_non_standard(mol2)
+    else:
+        return dm.to_inchikey(mol1) == dm.to_inchikey(mol2)
+
+
+def unique_id(mol: dm.Mol) -> Optional[str]:
+    """A datamol unique molecule ID.
+
+    The ID is an MD5 hash of the non-standard InChiKey provided
+    by `dm.to_inchikey_non_standard()`. It guarantees uniqueness for
+    different tautomeric forms of the same molecule.
+
+    Args:
+        mol: A molecule.
+    """
+    ik = dm.to_inchikey_non_standard(mol)
+
+    if ik is None:
+        return None
+
+    return hashlib.md5(ik.encode("utf-8")).hexdigest()
```

Comment:
I went with md5 here but happy to consider other hashing algorithm here (or no hash at all). Let me know.